{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6cbc06d3-997c-4f5b-9b66-4f658341c7da",
   "metadata": {},
   "outputs": [],
   "source": [
    "import deidHelpers as deid\n",
    "import large_image"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "219cf9c4-b981-46bd-862c-e5b16e1c971e",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: fastapi==0.111.0 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 1)) (0.111.0)\n",
      "Requirement already satisfied: fastapi-cli==0.0.3 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 2)) (0.0.3)\n",
      "Requirement already satisfied: imagecodecs==2024.1.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 3)) (2024.1.1)\n",
      "Requirement already satisfied: imagecodecs-numcodecs==2024.1.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 4)) (2024.1.1)\n",
      "Requirement already satisfied: jsonschema==4.22.0 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 5)) (4.22.0)\n",
      "Requirement already satisfied: jsonschema-specifications==2023.12.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 6)) (2023.12.1)\n",
      "Collecting large-image==1.28.1\n",
      "  Using cached large_image-1.28.1-py3-none-any.whl (94 kB)\n",
      "Requirement already satisfied: large-image-source-deepzoom==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 8)) (1.28.1)\n",
      "Requirement already satisfied: large-image-source-dicom==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 9)) (1.28.1)\n",
      "Requirement already satisfied: large-image-source-multi==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 10)) (1.28.1)\n",
      "Requirement already satisfied: large-image-source-nd2==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 11)) (1.28.1)\n",
      "Requirement already satisfied: large-image-source-pil==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 12)) (1.28.1)\n",
      "Requirement already satisfied: large-image-source-rasterio==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 13)) (1.28.1)\n",
      "Requirement already satisfied: large-image-source-test==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 14)) (1.28.1)\n",
      "Requirement already satisfied: large-image-source-tifffile==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 15)) (1.28.1)\n",
      "Requirement already satisfied: large-image-source-zarr==1.28.1 in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 16)) (1.28.1)\n",
      "Requirement already satisfied: girder_large_image in ./.pyenv/lib/python3.10/site-packages (from -r requirements.txt (line 17)) (1.28.1)\n",
      "Collecting pyvips\n",
      "  Downloading pyvips-2.2.3.tar.gz (56 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m56.6/56.6 kB\u001b[0m \u001b[31m2.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: httpx>=0.23.0 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (0.27.0)\n",
      "Requirement already satisfied: jinja2>=2.11.2 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (3.1.4)\n",
      "Requirement already satisfied: ujson!=4.0.2,!=4.1.0,!=4.2.0,!=4.3.0,!=5.0.0,!=5.1.0,>=4.0.1 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (5.10.0)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (2.7.3)\n",
      "Requirement already satisfied: orjson>=3.2.1 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (3.10.4)\n",
      "Requirement already satisfied: python-multipart>=0.0.7 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (0.0.9)\n",
      "Requirement already satisfied: uvicorn[standard]>=0.12.0 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (0.30.1)\n",
      "Requirement already satisfied: typing-extensions>=4.8.0 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (4.12.2)\n",
      "Requirement already satisfied: starlette<0.38.0,>=0.37.2 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (0.37.2)\n",
      "Requirement already satisfied: email_validator>=2.0.0 in ./.pyenv/lib/python3.10/site-packages (from fastapi==0.111.0->-r requirements.txt (line 1)) (2.1.1)\n",
      "Requirement already satisfied: typer>=0.12.3 in ./.pyenv/lib/python3.10/site-packages (from fastapi-cli==0.0.3->-r requirements.txt (line 2)) (0.12.3)\n",
      "Requirement already satisfied: numpy in ./.pyenv/lib/python3.10/site-packages (from imagecodecs==2024.1.1->-r requirements.txt (line 3)) (1.26.4)\n",
      "Requirement already satisfied: numcodecs in ./.pyenv/lib/python3.10/site-packages (from imagecodecs-numcodecs==2024.1.1->-r requirements.txt (line 4)) (0.12.1)\n",
      "Requirement already satisfied: referencing>=0.28.4 in ./.pyenv/lib/python3.10/site-packages (from jsonschema==4.22.0->-r requirements.txt (line 5)) (0.35.1)\n",
      "Requirement already satisfied: attrs>=22.2.0 in ./.pyenv/lib/python3.10/site-packages (from jsonschema==4.22.0->-r requirements.txt (line 5)) (23.2.0)\n",
      "Requirement already satisfied: rpds-py>=0.7.1 in ./.pyenv/lib/python3.10/site-packages (from jsonschema==4.22.0->-r requirements.txt (line 5)) (0.18.1)\n",
      "Requirement already satisfied: cachetools in ./.pyenv/lib/python3.10/site-packages (from large-image==1.28.1->-r requirements.txt (line 7)) (5.3.3)\n",
      "Requirement already satisfied: palettable in ./.pyenv/lib/python3.10/site-packages (from large-image==1.28.1->-r requirements.txt (line 7)) (3.3.3)\n",
      "Requirement already satisfied: Pillow in ./.pyenv/lib/python3.10/site-packages (from large-image==1.28.1->-r requirements.txt (line 7)) (10.3.0)\n",
      "Requirement already satisfied: wsidicom>=0.9.0 in ./.pyenv/lib/python3.10/site-packages (from large-image-source-dicom==1.28.1->-r requirements.txt (line 9)) (0.20.4)\n",
      "Requirement already satisfied: pyyaml in ./.pyenv/lib/python3.10/site-packages (from large-image-source-multi==1.28.1->-r requirements.txt (line 10)) (6.0.1)\n",
      "Requirement already satisfied: dask[array] in ./.pyenv/lib/python3.10/site-packages (from large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (2024.5.2)\n",
      "Requirement already satisfied: nd2[legacy]>=0.2.3 in ./.pyenv/lib/python3.10/site-packages (from large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (0.10.1)\n",
      "Requirement already satisfied: rasterio>=1.3 in ./.pyenv/lib/python3.10/site-packages (from large-image-source-rasterio==1.28.1->-r requirements.txt (line 13)) (1.3.10)\n",
      "Requirement already satisfied: packaging in ./.pyenv/lib/python3.10/site-packages (from large-image-source-rasterio==1.28.1->-r requirements.txt (line 13)) (24.1)\n",
      "Requirement already satisfied: tifffile[all] in ./.pyenv/lib/python3.10/site-packages (from large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (2024.5.22)\n",
      "Requirement already satisfied: zarr in ./.pyenv/lib/python3.10/site-packages (from large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (2.18.2)\n",
      "Requirement already satisfied: girder>=3.1.18 in ./.pyenv/lib/python3.10/site-packages (from girder_large_image->-r requirements.txt (line 17)) (3.2.3)\n",
      "Requirement already satisfied: girder-jobs>=3.0.3 in ./.pyenv/lib/python3.10/site-packages (from girder_large_image->-r requirements.txt (line 17)) (3.2.3)\n",
      "Requirement already satisfied: cffi>=1.0.0 in ./.pyenv/lib/python3.10/site-packages (from pyvips->-r requirements.txt (line 18)) (1.16.0)\n",
      "Requirement already satisfied: pycparser in ./.pyenv/lib/python3.10/site-packages (from cffi>=1.0.0->pyvips->-r requirements.txt (line 18)) (2.22)\n",
      "Requirement already satisfied: dnspython>=2.0.0 in ./.pyenv/lib/python3.10/site-packages (from email_validator>=2.0.0->fastapi==0.111.0->-r requirements.txt (line 1)) (2.6.1)\n",
      "Requirement already satisfied: idna>=2.0.0 in ./.pyenv/lib/python3.10/site-packages (from email_validator>=2.0.0->fastapi==0.111.0->-r requirements.txt (line 1)) (3.7)\n",
      "Requirement already satisfied: boto3 in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.34.122)\n",
      "Requirement already satisfied: botocore in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.34.122)\n",
      "Requirement already satisfied: CherryPy in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (18.9.0)\n",
      "Requirement already satisfied: click in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (8.1.7)\n",
      "Requirement already satisfied: click-plugins in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.1.1)\n",
      "Requirement already satisfied: dogpile.cache in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.3.3)\n",
      "Requirement already satisfied: filelock in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (3.14.0)\n",
      "Requirement already satisfied: Mako in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.3.5)\n",
      "Requirement already satisfied: passlib[bcrypt,totp] in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.7.4)\n",
      "Requirement already satisfied: pymongo<4,>=3.6 in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (3.13.0)\n",
      "Requirement already satisfied: psutil in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (5.9.8)\n",
      "Requirement already satisfied: pyOpenSSL in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (24.1.0)\n",
      "Requirement already satisfied: python-dateutil in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (2.9.0.post0)\n",
      "Requirement already satisfied: pytz in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (2024.1)\n",
      "Requirement already satisfied: requests in ./.pyenv/lib/python3.10/site-packages (from girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (2.32.3)\n",
      "Requirement already satisfied: certifi in ./.pyenv/lib/python3.10/site-packages (from httpx>=0.23.0->fastapi==0.111.0->-r requirements.txt (line 1)) (2024.6.2)\n",
      "Requirement already satisfied: anyio in ./.pyenv/lib/python3.10/site-packages (from httpx>=0.23.0->fastapi==0.111.0->-r requirements.txt (line 1)) (4.4.0)\n",
      "Requirement already satisfied: sniffio in ./.pyenv/lib/python3.10/site-packages (from httpx>=0.23.0->fastapi==0.111.0->-r requirements.txt (line 1)) (1.3.1)\n",
      "Requirement already satisfied: httpcore==1.* in ./.pyenv/lib/python3.10/site-packages (from httpx>=0.23.0->fastapi==0.111.0->-r requirements.txt (line 1)) (1.0.5)\n",
      "Requirement already satisfied: h11<0.15,>=0.13 in ./.pyenv/lib/python3.10/site-packages (from httpcore==1.*->httpx>=0.23.0->fastapi==0.111.0->-r requirements.txt (line 1)) (0.14.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in ./.pyenv/lib/python3.10/site-packages (from jinja2>=2.11.2->fastapi==0.111.0->-r requirements.txt (line 1)) (2.1.5)\n",
      "Requirement already satisfied: ome-types>=0.5.1 in ./.pyenv/lib/python3.10/site-packages (from nd2[legacy]>=0.2.3->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (0.5.1.post1)\n",
      "Requirement already satisfied: resource-backed-dask-array in ./.pyenv/lib/python3.10/site-packages (from nd2[legacy]>=0.2.3->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (0.1.0)\n",
      "Requirement already satisfied: pydantic-core==2.18.4 in ./.pyenv/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi==0.111.0->-r requirements.txt (line 1)) (2.18.4)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in ./.pyenv/lib/python3.10/site-packages (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi==0.111.0->-r requirements.txt (line 1)) (0.7.0)\n",
      "Requirement already satisfied: cligj>=0.5 in ./.pyenv/lib/python3.10/site-packages (from rasterio>=1.3->large-image-source-rasterio==1.28.1->-r requirements.txt (line 13)) (0.7.2)\n",
      "Requirement already satisfied: snuggs>=1.4.1 in ./.pyenv/lib/python3.10/site-packages (from rasterio>=1.3->large-image-source-rasterio==1.28.1->-r requirements.txt (line 13)) (1.4.7)\n",
      "Requirement already satisfied: setuptools in ./.pyenv/lib/python3.10/site-packages (from rasterio>=1.3->large-image-source-rasterio==1.28.1->-r requirements.txt (line 13)) (65.5.0)\n",
      "Requirement already satisfied: affine in ./.pyenv/lib/python3.10/site-packages (from rasterio>=1.3->large-image-source-rasterio==1.28.1->-r requirements.txt (line 13)) (2.4.0)\n",
      "Requirement already satisfied: rich>=10.11.0 in ./.pyenv/lib/python3.10/site-packages (from typer>=0.12.3->fastapi-cli==0.0.3->-r requirements.txt (line 2)) (13.7.1)\n",
      "Requirement already satisfied: shellingham>=1.3.0 in ./.pyenv/lib/python3.10/site-packages (from typer>=0.12.3->fastapi-cli==0.0.3->-r requirements.txt (line 2)) (1.5.4)\n",
      "Requirement already satisfied: uvloop!=0.15.0,!=0.15.1,>=0.14.0 in ./.pyenv/lib/python3.10/site-packages (from uvicorn[standard]>=0.12.0->fastapi==0.111.0->-r requirements.txt (line 1)) (0.19.0)\n",
      "Requirement already satisfied: watchfiles>=0.13 in ./.pyenv/lib/python3.10/site-packages (from uvicorn[standard]>=0.12.0->fastapi==0.111.0->-r requirements.txt (line 1)) (0.22.0)\n",
      "Requirement already satisfied: python-dotenv>=0.13 in ./.pyenv/lib/python3.10/site-packages (from uvicorn[standard]>=0.12.0->fastapi==0.111.0->-r requirements.txt (line 1)) (1.0.1)\n",
      "Requirement already satisfied: httptools>=0.5.0 in ./.pyenv/lib/python3.10/site-packages (from uvicorn[standard]>=0.12.0->fastapi==0.111.0->-r requirements.txt (line 1)) (0.6.1)\n",
      "Requirement already satisfied: websockets>=10.4 in ./.pyenv/lib/python3.10/site-packages (from uvicorn[standard]>=0.12.0->fastapi==0.111.0->-r requirements.txt (line 1)) (12.0)\n",
      "Requirement already satisfied: fsspec<2025.0.0,>=2024.2.0 in ./.pyenv/lib/python3.10/site-packages (from wsidicom>=0.9.0->large-image-source-dicom==1.28.1->-r requirements.txt (line 9)) (2024.6.0)\n",
      "Requirement already satisfied: universal-pathlib<0.3.0,>=0.2.1 in ./.pyenv/lib/python3.10/site-packages (from wsidicom>=0.9.0->large-image-source-dicom==1.28.1->-r requirements.txt (line 9)) (0.2.2)\n",
      "Requirement already satisfied: pydicom>=2.1.0 in ./.pyenv/lib/python3.10/site-packages (from wsidicom>=0.9.0->large-image-source-dicom==1.28.1->-r requirements.txt (line 9)) (2.4.4)\n",
      "Requirement already satisfied: marshmallow<4.0.0,>=3.20.1 in ./.pyenv/lib/python3.10/site-packages (from wsidicom>=0.9.0->large-image-source-dicom==1.28.1->-r requirements.txt (line 9)) (3.21.3)\n",
      "Requirement already satisfied: dicomweb-client<0.60.0,>=0.59.1 in ./.pyenv/lib/python3.10/site-packages (from wsidicom>=0.9.0->large-image-source-dicom==1.28.1->-r requirements.txt (line 9)) (0.59.1)\n",
      "Requirement already satisfied: toolz>=0.10.0 in ./.pyenv/lib/python3.10/site-packages (from dask[array]->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (0.12.1)\n",
      "Requirement already satisfied: cloudpickle>=1.5.0 in ./.pyenv/lib/python3.10/site-packages (from dask[array]->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (3.0.0)\n",
      "Requirement already satisfied: importlib-metadata>=4.13.0 in ./.pyenv/lib/python3.10/site-packages (from dask[array]->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (7.1.0)\n",
      "Requirement already satisfied: partd>=1.2.0 in ./.pyenv/lib/python3.10/site-packages (from dask[array]->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (1.4.2)\n",
      "Requirement already satisfied: defusedxml in ./.pyenv/lib/python3.10/site-packages (from tifffile[all]->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (0.7.1)\n",
      "Requirement already satisfied: matplotlib in ./.pyenv/lib/python3.10/site-packages (from tifffile[all]->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (3.9.0)\n",
      "Requirement already satisfied: lxml in ./.pyenv/lib/python3.10/site-packages (from tifffile[all]->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (5.2.2)\n",
      "Requirement already satisfied: asciitree in ./.pyenv/lib/python3.10/site-packages (from zarr->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (0.3.3)\n",
      "Requirement already satisfied: fasteners in ./.pyenv/lib/python3.10/site-packages (from zarr->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (0.19)\n",
      "Requirement already satisfied: exceptiongroup>=1.0.2 in ./.pyenv/lib/python3.10/site-packages (from anyio->httpx>=0.23.0->fastapi==0.111.0->-r requirements.txt (line 1)) (1.2.1)\n",
      "Requirement already satisfied: retrying>=1.3.3 in ./.pyenv/lib/python3.10/site-packages (from dicomweb-client<0.60.0,>=0.59.1->wsidicom>=0.9.0->large-image-source-dicom==1.28.1->-r requirements.txt (line 9)) (1.3.4)\n",
      "Requirement already satisfied: zipp>=0.5 in ./.pyenv/lib/python3.10/site-packages (from importlib-metadata>=4.13.0->dask[array]->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (3.19.2)\n",
      "Requirement already satisfied: xsdata<24.4,>=23.6 in ./.pyenv/lib/python3.10/site-packages (from ome-types>=0.5.1->nd2[legacy]>=0.2.3->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (24.3.1)\n",
      "Requirement already satisfied: pydantic-compat>=0.1.0 in ./.pyenv/lib/python3.10/site-packages (from ome-types>=0.5.1->nd2[legacy]>=0.2.3->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (0.1.2)\n",
      "Requirement already satisfied: locket in ./.pyenv/lib/python3.10/site-packages (from partd>=1.2.0->dask[array]->large-image-source-nd2==1.28.1->-r requirements.txt (line 11)) (1.0.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in ./.pyenv/lib/python3.10/site-packages (from requests->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (3.3.2)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in ./.pyenv/lib/python3.10/site-packages (from requests->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (2.2.1)\n",
      "Requirement already satisfied: markdown-it-py>=2.2.0 in ./.pyenv/lib/python3.10/site-packages (from rich>=10.11.0->typer>=0.12.3->fastapi-cli==0.0.3->-r requirements.txt (line 2)) (3.0.0)\n",
      "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in ./.pyenv/lib/python3.10/site-packages (from rich>=10.11.0->typer>=0.12.3->fastapi-cli==0.0.3->-r requirements.txt (line 2)) (2.18.0)\n",
      "Requirement already satisfied: pyparsing>=2.1.6 in ./.pyenv/lib/python3.10/site-packages (from snuggs>=1.4.1->rasterio>=1.3->large-image-source-rasterio==1.28.1->-r requirements.txt (line 13)) (3.1.2)\n",
      "Requirement already satisfied: s3transfer<0.11.0,>=0.10.0 in ./.pyenv/lib/python3.10/site-packages (from boto3->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (0.10.1)\n",
      "Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in ./.pyenv/lib/python3.10/site-packages (from boto3->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.0.1)\n",
      "Requirement already satisfied: six>=1.5 in ./.pyenv/lib/python3.10/site-packages (from python-dateutil->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.16.0)\n",
      "Requirement already satisfied: zc.lockfile in ./.pyenv/lib/python3.10/site-packages (from CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (3.0.post1)\n",
      "Requirement already satisfied: cheroot>=8.2.1 in ./.pyenv/lib/python3.10/site-packages (from CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (10.0.1)\n",
      "Requirement already satisfied: more-itertools in ./.pyenv/lib/python3.10/site-packages (from CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (10.3.0)\n",
      "Requirement already satisfied: portend>=2.1.1 in ./.pyenv/lib/python3.10/site-packages (from CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (3.2.0)\n",
      "Requirement already satisfied: jaraco.collections in ./.pyenv/lib/python3.10/site-packages (from CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (5.0.1)\n",
      "Requirement already satisfied: decorator>=4.0.0 in ./.pyenv/lib/python3.10/site-packages (from dogpile.cache->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (5.1.1)\n",
      "Requirement already satisfied: stevedore>=3.0.0 in ./.pyenv/lib/python3.10/site-packages (from dogpile.cache->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (5.2.0)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in ./.pyenv/lib/python3.10/site-packages (from matplotlib->tifffile[all]->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (4.53.0)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in ./.pyenv/lib/python3.10/site-packages (from matplotlib->tifffile[all]->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (1.2.1)\n",
      "Requirement already satisfied: kiwisolver>=1.3.1 in ./.pyenv/lib/python3.10/site-packages (from matplotlib->tifffile[all]->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (1.4.5)\n",
      "Requirement already satisfied: cycler>=0.10 in ./.pyenv/lib/python3.10/site-packages (from matplotlib->tifffile[all]->large-image-source-tifffile==1.28.1->-r requirements.txt (line 15)) (0.12.1)\n",
      "Requirement already satisfied: bcrypt>=3.1.0 in ./.pyenv/lib/python3.10/site-packages (from passlib[bcrypt,totp]->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (4.1.3)\n",
      "Requirement already satisfied: cryptography in ./.pyenv/lib/python3.10/site-packages (from passlib[bcrypt,totp]->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (42.0.8)\n",
      "Requirement already satisfied: jaraco.functools in ./.pyenv/lib/python3.10/site-packages (from cheroot>=8.2.1->CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (4.0.1)\n",
      "Requirement already satisfied: mdurl~=0.1 in ./.pyenv/lib/python3.10/site-packages (from markdown-it-py>=2.2.0->rich>=10.11.0->typer>=0.12.3->fastapi-cli==0.0.3->-r requirements.txt (line 2)) (0.1.2)\n",
      "Requirement already satisfied: tempora>=1.8 in ./.pyenv/lib/python3.10/site-packages (from portend>=2.1.1->CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (5.5.1)\n",
      "Requirement already satisfied: pbr!=2.1.0,>=2.0.0 in ./.pyenv/lib/python3.10/site-packages (from stevedore>=3.0.0->dogpile.cache->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (6.0.0)\n",
      "Requirement already satisfied: jaraco.text in ./.pyenv/lib/python3.10/site-packages (from jaraco.collections->CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (3.12.0)\n",
      "Requirement already satisfied: jaraco.context>=4.1 in ./.pyenv/lib/python3.10/site-packages (from jaraco.text->jaraco.collections->CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (5.3.0)\n",
      "Requirement already satisfied: autocommand in ./.pyenv/lib/python3.10/site-packages (from jaraco.text->jaraco.collections->CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (2.2.2)\n",
      "Requirement already satisfied: inflect in ./.pyenv/lib/python3.10/site-packages (from jaraco.text->jaraco.collections->CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (7.2.1)\n",
      "Requirement already satisfied: backports.tarfile in ./.pyenv/lib/python3.10/site-packages (from jaraco.context>=4.1->jaraco.text->jaraco.collections->CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (1.2.0)\n",
      "Requirement already satisfied: typeguard>=4.0.1 in ./.pyenv/lib/python3.10/site-packages (from inflect->jaraco.text->jaraco.collections->CherryPy->girder>=3.1.18->girder_large_image->-r requirements.txt (line 17)) (4.3.0)\n",
      "Installing collected packages: large-image, pyvips\n",
      "  Attempting uninstall: large-image\n",
      "    Found existing installation: large-image 1.29.0\n",
      "    Uninstalling large-image-1.29.0:\n",
      "      Successfully uninstalled large-image-1.29.0\n",
      "\u001b[33m  DEPRECATION: pyvips is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559\u001b[0m\u001b[33m\n",
      "\u001b[0m  Running setup.py install for pyvips ... \u001b[?25ldone\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "large-image-source-tiff 1.29.0 requires large-image>=1.29.0, but you have large-image 1.28.1 which is incompatible.\u001b[0m\u001b[31m\n",
      "\u001b[0m\u001b[?25hSuccessfully installed large-image-1.28.1 pyvips-2.2.3\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.0.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m24.0\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install -r requirements.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a924e3c5-d1cd-4edf-877d-8714550f9729",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample slide data/CMU-2-backup.svs\n",
      "In tempdir\n"
     ]
    }
   ],
   "source": [
    "deid.deid_workflow()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0cb248e-0135-4290-bd76-c5b534a49049",
   "metadata": {},
   "outputs": [],
   "source": [
    "ts = large_image.open('./data/CMU-2-backup.svs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0da9e04d-0f60-46db-a6b0-1bf58f688bfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "import large_image"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dad0f812-eb9e-4bce-bb9a-1f0ac3ddfca1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
